services/mediastoredata/src/it/java/software/amazon/awssdk/services/mediastoredata/TransferEncodingChunkedIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.mediastoredata;
+
+import static software.amazon.awssdk.utils.FunctionalUtils.invokeSafely;
+
+import io.reactivex.Flowable;
+import java.io.ByteArrayInputStream;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.reactivestreams.Subscriber;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.http.nio.netty.NettyNioAsyncHttpClient;
+import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.services.mediastore.MediaStoreClient;
+import software.amazon.awssdk.services.mediastore.model.Container;
+import software.amazon.awssdk.services.mediastore.model.ContainerStatus;
+import software.amazon.awssdk.services.mediastore.model.DescribeContainerResponse;
+import software.amazon.awssdk.services.mediastoredata.model.DeleteObjectRequest;
+import software.amazon.awssdk.services.mediastoredata.model.PutObjectRequest;
+import software.amazon.awssdk.testutils.Waiter;
+import software.amazon.awssdk.testutils.service.AwsIntegrationTestBase;
+
+public class TransferEncodingChunkedIntegrationTest extends AwsIntegrationTestBase {
```

Comment:
Can we add a comment mentioning this integ test is to verify chunked transfer encoding functionalities for all supported HTTP clients and should not be deleted?  because it's possible we'll forget about this and delete this test down the line.

Reply:
added comment